packages/compiler/core/program.ts

```diff
@@ -986,9 +986,7 @@ export async function compile(
     );
 
     if (actual.mainFile !== expected && MANIFEST.version !== actual.manifest.version) {
-      // we have resolved node_modules/@typespec/compiler/dist/core/index.js and we want to get
-      // to the shim executable node_modules/.bin/tsp-server
-      const betterTypeSpecServerPath = resolvePath(actual.path, ".bin/tsp-server");
+      const betterTypeSpecServerPath = actual.path;
```

Comment:
Does this actually improve the error, though?  Seems like it just reverts back to pointing back to the `.js` file instead.  If it does improve things, maybe we can just drop the `betterTypeSpecServerPath` and use `actual.path` instead?

Reply:
The comment above is actually incorrect, it was not updated last times actual.path is pointing to the compiler/ folder